### ISSUE TITLE
Windows CoApp Registration failed with custom install location

### ISSUE THREAD

**Issue by gaveitatry2** [aclap-dev/vdhcoapp#201]
See: https://github.com/aclap-dev/vdhcoapp/issues/201#issuecomment-1888399384

**Comment by gaveitatry2** [comment]
So annoying. The registry changes didn't work. I don't want to re-install my entire browser or re-install anything. Grrrrrr.

**Comment by gaveitatry2** [comment]
![firefox_nqx6m6oraq](https://github.com/aclap-dev/vdhcoapp/assets/75463665/68aadb34-9de6-432c-bf53-2415a58ff45c)


![explorer_auDP4K4RDg](https://github.com/aclap-dev/vdhcoapp/assets/75463665/ea9c6bec-2084-4dbe-a4e1-07560aaee74a)

How is it my fault that "No such native application net.downloadhelper.coapp". It's not my fault that the co-op installer changed the application name and it is now called vdhcoapp.exe in the 'DownloadHelper CoApp' folder.

**Comment by gaveitatry2** [comment]
Tried the 32-bit Windows CoApp installer just now. That didn't work either.

**Comment by gaveitatry2** [comment]
I reverted back to 8.0.0.6, disabled automatic updates, and downloaded CoApp 1.6.3 from some third party site (because I didn't know where to get it from). Everything working again. Hopefully a fix will come for the newer version so I can be up-to-date.

**Comment by gaveitatry2** [comment]
@paulrouget 
 
Yes, what you said is correct.

**THIS DOES NOT WORK:**

Mozilla Firefox (Portable) installed at:
"G:\Portable Apps\PortableApps\FirefoxPortable"

Video DownloadHelper 8.1.4.0

CoApp 2.0.10 installed at (I tried both Windows 64-bit and 32-bit versions):
"G:\Portable Apps\PortableApps\FirefoxPortable\DownloadHelper CoApp"

**THIS WORKS:**

Mozilla Firefox (Portable) installed at:
"G:\Portable Apps\PortableApps\FirefoxPortable"

Video DownloadHelper 8.0.0.6

CoApp 1.6.3 installed at:
"G:\Portable Apps\PortableApps\FirefoxPortable\net.downloadhelper.coapp"

Also, no Antivirus blocking anything.

I am just going to use the older version until you figure things out.

**Comment by paulrouget** [comment]
If that's possible, do you mind trying to install the coapp v2 into its default location?

I would like to understand if it's an issue with the custom install location or the existence of a previously installe V1 coapp (which I think is the issue here).

Thanks for helping out.

**Comment by gaveitatry2** [comment]
@paulrouget 

Please ell me what the default location is. I had already replaced it with my desired location. Also, it won't work if the default location is in some folder for the desktop version of Mozilla Firefox as I don't have a desktop version of the browser installed.

**Comment by gaveitatry2** [comment]
@paulrouget 

I did as you asked. It did not work.

I updated VDH. I installed CoApp to "C:\Program Files\DownloadHelper CoApp". I restarted my browser.

![explorer_NpFWs4lJUN](https://github.com/aclap-dev/vdhcoapp/assets/75463665/5a949120-9b2a-4b40-bb59-483c0392fbcf)

**Comment by paulrouget** [comment]
Can you check this for me:

In your registry (regedit) can you look up this value for me:

```
HKEY_LOCAL_MACHINE\SOFTWARE\Mozilla\NativeMessagingHosts
HKEY_CURRENT_USER\Software\Mozilla\NativeMessagingHosts
```

And maybe:

```
HKEY_LOCAL_MACHINE\SOFTWARE\WOW****\Mozilla\NativeMessagingHosts (**** could be anything)
```

I need to know the "Data" section (it's a path, like c:\Program Files\…).

This is what Firefox looks at to know the position of the JSON file.

My guess is that `HKEY_LOCAL_MACHINE\SOFTWARE\Mozilla\NativeMessagingHosts` will point to the new coapp and `HKEY_CURRENT_USER\Software\Mozilla\NativeMessagingHosts` to the old coapp.

Let me know.

Thank you again.

**Comment by gaveitatry2** [comment]
@paulrouget 

Do I have to reinstall the newest VDH and CoApp first before I check? I already reverted back to the old version. Also, do I have to install CoApp in the default location even though I want the custom location?

**Comment by paulrouget** [comment]
Please install the CoApp v2 (anywhere you want) and Addon 8.1.

It will not be recognized as per your bug. Look at the registry like I mentioned.

Check that the values point to the right json file.

---

After than, remove CoApp v2, install CoApp v1.

Does the Addon recognize the CoApp v1? (I think it will, and it will tell you to update)

Look at the registry again.

---

Thank you!

**Comment by gaveitatry2** [comment]
@paulrouget 

Okay, I did as you asked. I also restarted the browser and Registry Editor each time.

**Video DownloadHelper 8.1.4.0
Companion App 2.0.10 (installed in Firefox Portable location)**

Computer\HKEY_CURRENT_USER\SOFTWARE\Mozilla\NativeMessagingHosts\DownloadHelper CoApp
Data: (value not set)

Computer\HKEY_CURRENT_USER\SOFTWARE\Mozilla\NativeMessagingHosts\net.downloadhelper.coapp
Data: G:\Portable Apps\PortableApps\FirefoxPortable\net.downloadhelper.coapp\firefox.net.downloadhelper.coapp.json

Computer\HKEY_CURRENT_USER\SOFTWARE\Wow6432Node
Note: There is nothing here except a Microsoft folder (with Active Setup and Windows subfolders). There is no Mozilla\NativeMessagingHosts folder.

Computer\HKEY_LOCAL_MACHINE\SOFTWARE\Mozilla\NativeMessagingHosts\DownloadHelper CoApp
Data: (value not set)

Computer\HKEY_LOCAL_MACHINE\SOFTWARE\Mozilla\NativeMessagingHosts\net.downloadhelper.coapp
Data: G:\Portable Apps\PortableApps\FirefoxPortable\net.downloadhelper.coapp\firefox.net.downloadhelper.coapp.json

Note: I then just deleted the DownloadHelper CoApp folder. I did not use the uninstall.exe inside (never have before). And then I installed CoApp 1.6.3. PS - I did not know where to get CoApp 1.6.3 so I downloaded from wingetgui, and it did not prompt me on where I wanted to install. It just installed to the last install location. Maybe because I needed a slightly newer version like 1.6.3.1 (or whatever the version number is) that would prompt me on install location.

**Video DownloadHelper 8.1.4.0
Companion App 1.6.3 (installed in Firefox Portable location)**

"Companion App outdated - please update". It won't let me download any videos, which is why I am tempted to install Video DownloadHelper 8.0.0.6 to pair with CoApp 1.6.3, and then get the registry data. But you didn't ask me to, so I didn't.
 
Computer\HKEY_CURRENT_USER\SOFTWARE\Mozilla\NativeMessagingHosts\DownloadHelper CoApp
Data: (value not set)

Computer\HKEY_CURRENT_USER\SOFTWARE\Mozilla\NativeMessagingHosts\net.downloadhelper.coapp
Data: G:\Portable Apps\PortableApps\FirefoxPortable\net.downloadhelper.coapp\firefox.net.downloadhelper.coapp.json

Computer\HKEY_CURRENT_USER\SOFTWARE\Wow6432Node
Note: There is nothing here except a Microsoft folder (with Active Setup and Windows subfolders). There is no Mozilla\NativeMessagingHosts folder.

Computer\HKEY_LOCAL_MACHINE\SOFTWARE\Mozilla\NativeMessagingHosts\DownloadHelper CoApp
Data: (value not set)

Computer\HKEY_LOCAL_MACHINE\SOFTWARE\Mozilla\NativeMessagingHosts\net.downloadhelper.coapp
Data: G:\Portable Apps\PortableApps\FirefoxPortable\net.downloadhelper.coapp\firefox.net.downloadhelper.coapp.json

**Video DownloadHelper 8.1.4.0
Companion App 2.0.10 (installed in Firefox Portable location)
After using uninstaller (unins000.exe) on Companion App 1.6.3.**

I had never ever used the CoApp uninstallers before. I noticed it just now and decided to use it to see if it made a difference. It only deleted a net.downloadhelper.coapp folder in HKEY_LOCAL_MACHINE, but didn't delete the same folder in HKEY_CURRENT_USER.

"Companion App not installed
Checking companion app returned: No such native application net.downloadhelper.coapp."

Computer\HKEY_CURRENT_USER\SOFTWARE\Mozilla\NativeMessagingHosts\DownloadHelper CoApp
Data: (value not set)

Computer\HKEY_CURRENT_USER\SOFTWARE\Mozilla\NativeMessagingHosts\net.downloadhelper.coapp
Data: G:\Portable Apps\PortableApps\FirefoxPortable\net.downloadhelper.coapp\firefox.net.downloadhelper.coapp.json

Computer\HKEY_CURRENT_USER\SOFTWARE\Wow6432Node
Note: There is nothing here except a Microsoft folder (with Active Setup and Windows subfolders). There is no Mozilla\NativeMessagingHosts folder.

Computer\HKEY_LOCAL_MACHINE\SOFTWARE\Mozilla\NativeMessagingHosts\DownloadHelper CoApp
Data: (value not set)

Computer\HKEY_LOCAL_MACHINE\SOFTWARE\Mozilla\NativeMessagingHosts\net.downloadhelper.coapp
NO LONGER EXISTS AFTER HAVING USED THE UNINSTALLER, even though Computer\HKEY_CURRENT_USER\SOFTWARE\Mozilla\NativeMessagingHosts\net.downloadhelper.coapp still exists after having used the uninstaller.

**This is something else that I wanted to share.**

"Companion App not installed
Checking companion app returned: No such native application net.downloadhelper.coapp."

CoApp 1.6.3 is installed in a folder called net.downloadhelper.coapp, but CoApp 2.0.10 is installed in a folder called DownloadHelper CoApp. So why is Video DownloadHelper 8.1.4.0 looking for net.downloadhelper.coapp?

**Comment by paulrouget** [comment]
You're not the only one with this issue. I'm actively trying to understand what's going on here. I keep you updated.

**Comment by paulrouget** [comment]
@gaveitatry2 

You confirm that you use a **64 bits Windows**, with the **64 bits CoApp V2**, and you use **Firefox 64 bits**?

If you run in a prompt this command, what does it say?
```
reg query HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Mozilla\NativeMessagingHosts\net.downloadhelper.coapp
```

**Comment by gaveitatry2** [comment]
@paulrouget 

Yes. 64 bit Windows, 64 bit CoApp v2, 64 bit Mozilla Firefox (Portable). Windows 10.

If I copy and paste that command in Command Prompt while using VDH 8.0.0.6 and CoApp v1 (because this combination is the only thing that works for me), I get the following response:
ERROR: The system was unable to find the specified registry key or value.

Do you need me to test it out with VDH 8.1.4.0 and CoApp v2 and also VDH 8.1.4.0 and CoApp v1 as well?

**Comment by paulrouget** [comment]
Thank you for trying this out.

With CoApp V1, what does this say:

```
reg query HKEY_LOCAL_MACHINE\SOFTWARE\Mozilla\NativeMessagingHosts\net.downloadhelper.coapp
reg query HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Mozilla\NativeMessagingHosts\net.downloadhelper.coapp
```

Then, install the V2, and tell me again what these 2 commands say?

I think maybe Portable Firefox is built in 32 bits mode.

So maybe coapp v2 32 bits will work for you.

**Comment by paulrouget** [comment]
This is really strange that your 64bits portable firefox reads the 32bits registry.

Do you confirm that if you copy (with regedit) the value of

```
HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Mozilla\NativeMessagingHosts\net.downloadhelper.coapp
```

into

```
HKEY_LOCAL_MACHINE\SOFTWARE\Mozilla\NativeMessagingHosts\net.downloadhelper.coapp
```

Then the coapp is recognized?

**Comment by gaveitatry2** [comment]
@paulrouget 

**VDH 8.1.4.0 and CoApp v2 (64-bit) (in Firefox Portable folder)**

"Companion App not installed."

HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Mozilla\NativeMessagingHosts\net.downloadhelper.coapp
In Registry Editor, I copied the following from Data:
G:\Portable Apps\PortableApps\FirefoxPortable\DownloadHelper CoApp\mozilla.json

HKEY_LOCAL_MACHINE\SOFTWARE\Mozilla\NativeMessagingHosts\net.downloadhelper.coapp
Does not exist. But the following exists.
HKEY_LOCAL_MACHINE\SOFTWARE\Mozilla\NativeMessagingHosts\DownloadHelper CoApp
I pasted the following into Data:
G:\Portable Apps\PortableApps\FirefoxPortable\DownloadHelper CoApp\mozilla.json

I restarted the browser. Still "Companion App not installed."

Is it possible that the space in "DownloadHelper CoApp" creates problems? Previously, with CoApp v1, it was "net.downloadhelper.coapp" which had no spaces. I don't know. I am just throwing out random ideas.

**Comment by paulrouget** [comment]
> If it is looking for CoApp v2, it should say "no such native application DownloadHelper CoApp"

No. That's the ID of the app. And it's correct. So the Regedit key should be:

```
HKEY_LOCAL_MACHINE\SOFTWARE\Mozilla\NativeMessagingHosts\net.downloadhelper.coapp
```

If you set `HKEY_LOCAL_MACHINE\SOFTWARE\Mozilla\NativeMessagingHosts\net.downloadhelper.coapp` to the mozilla.json path, does it work?

**Comment by paulrouget** [comment]
Ok so apparently some custom builds of Firefox (Portable or Waterfox), even in 64bits mode, use the 32 bits registry db.

The coapp would then need to enforce that somehow. I'll be looking at it.

**Comment by gaveitatry2** [comment]
@paulrouget 

I believe the following is accurate, but I might have made some mistakes.

COAPP v1 installs the following:
HKEY_LOCAL_MACHINE\SOFTWARE\Mozilla\NativeMessagingHosts\net.downloadhelper.coapp
And the json path is set to:
G:\Portable Apps\PortableApps\FirefoxPortable\net.downloadhelper.coapp\firefox.net.downloadhelper.coapp.json
COAPP v1 also installs the following COAPP folder to ProgramFiles (or whatever location I choose):
net.downloadhelper.coapp

COAPP v2 installs the following:
HKEY_LOCAL_MACHINE\SOFTWARE\Mozilla\NativeMessagingHosts\DownloadHelper CoApp
And the json path is set to:
nothing
COAPP v2 also installs the following COAPP folder ProgramFiles (or whatever location I choose):
DownloadHelper CoApp
COAPP v2 also results in the following message:
No such native application net.downloadhelper.coapp.


If I use the COAPP v1 uninstaller before installing COAPP v2, this disappears:
HKEY_LOCAL_MACHINE\SOFTWARE\Mozilla\NativeMessagingHosts\net.downloadhelper.coapp
So for it to exist on COAPP v2, I have to install COAPP v1 before installing COAPP v2, and I have to be sure not to use the COAPP v1 uninstaller.

It seems that you want me to change the json path for:
HKEY_LOCAL_MACHINE\SOFTWARE\Mozilla\NativeMessagingHosts\net.downloadhelper.coapp
(which COAPP v2 does not install, but only exists because it is leftover from COAPP v1)
from:
G:\Portable Apps\PortableApps\FirefoxPortable\net.downloadhelper.coapp\firefox.net.downloadhelper.coapp.json
to:
G:\Portable Apps\PortableApps\FirefoxPortable\DownloadHelper CoApp\mozilla.json
which I got from:
HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Mozilla\NativeMessagingHosts\net.downloadhelper.coapp

As a result from this and previous tests, the following:
HKEY_LOCAL_MACHINE\SOFTWARE\Mozilla\NativeMessagingHosts\DownloadHelper CoApp
HKEY_LOCAL_MACHINE\SOFTWARE\Mozilla\NativeMessagingHosts\net.downloadhelper.coapp
are now both set to:
G:\Portable Apps\PortableApps\FirefoxPortable\DownloadHelper CoApp\mozilla.json

But I still get:
Companion App not installed.
No such native application net.downloadhelper.coapp.

**Comment by gaveitatry2** [comment]
This is what COAPP v1 installs in the "net.downloadhelper.coapp" folder:

![explorer_ojmNnZzsu8](https://github.com/aclap-dev/vdhcoapp/assets/75463665/3870b4ba-37e7-485a-9e88-82e960f58643)

This is what COAPP v2 installs in the "DownloadHelper CoApp" folder:

![explorer_fAXxpoURqr](https://github.com/aclap-dev/vdhcoapp/assets/75463665/3b4ee1cc-dd28-4ecd-9b0f-b5dba44b39f5)

Is it possible that VDH is not detecting COAPP v2 json file because the "DownloadHelper CoApp" folder is missing the bin and converter subfolders that COAPP v1's "net.downloadhelper.coapp" folder has?

**Comment by paulrouget** [comment]
This is confusing me a lot… I don't understand why you get `HKEY_LOCAL_MACHINE\SOFTWARE\Mozilla\NativeMessagingHosts\DownloadHelper CoApp`. As far as I know, where never create that key. It's very strange.

If you start a CMD prompt and type:

```
G:\Portable Apps\PortableApps\FirefoxPortable\DownloadHelper CoApp\vdhcoapp --info
```

Does it say anything?

You can also directly contact me on Discord: https://discord.gg/ygCp6zPE if that's easier for you.

**Comment by gaveitatry2** [comment]
@paulrouget 

If I open up Command Prompt from the Windows 10 Start menu and type:
G:\Portable Apps\PortableApps\FirefoxPortable\DownloadHelper CoApp\vdhcoapp --info

This is the response:
'G:\Portable' is not recognized as an internal or external command, operable program or batch file.

"This is confusing me a lot… I don't understand why you get HKEY_LOCAL_MACHINE\SOFTWARE\Mozilla\NativeMessagingHosts\DownloadHelper CoApp. As far as I know, we never create that key. It's very strange."

What is so strange about it? I already mentioned four times that COAPP v2 installs to a "DownloadHelper CoApp" folder instead of how COAPP v1 installs to a "net.downloadhelper.coapp" folder. Obviously, someone wanted to change "net.downloadhelper.coapp" to "DownloadHelper CoApp" so it makes sense to not only change the folder name but also the registry name. Why does this not make sense? Aren't you the one who made these change? If not, what don't you consult with THAT person and maybe you guys can reference this GitHub thread and figure out what is going on.

I don't want to chat on Discord. I will continue using this GitHub Issue thread.

**Comment by gaveitatry2** [comment]
@paulrouget 

I don't know anything about Command Prompt, but I decided to try extra quotation marks (because of the spaces).
I typed this instead:
"""G:\Portable Apps\PortableApps\FirefoxPortable\DownloadHelper CoApp\vdhcoapp""" --info

This was the response:

```
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.10",
  "binary": "G:\\Portable Apps\\PortableApps\\FirefoxPortable\\DownloadHelper CoApp\\vdhcoapp.exe",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "windows",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "C:\\Users\\Brian",
  "converterBinary": "G:\\Portable Apps\\PortableApps\\FirefoxPortable\\DownloadHelper CoApp\\ffmpeg.exe",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "6.0"
}
```

**Comment by paulrouget** [comment]
> What is so strange about it? I already mentioned four times that COAPP v2 installs to a "DownloadHelper CoApp" folder instead of how COAPP v1 installs to a "net.downloadhelper.coapp" folder. Obviously, someone wanted to change "net.downloadhelper.coapp" to "DownloadHelper CoApp" so it makes sense to not only change the folder name but also the registry name. Why does this not make sense? Aren't you the one who made these change? If not, what don't you consult with THAT person and maybe you guys can reference this GitHub thread and figure out what is going on.

I made these changes myself. I'm the main developer of the CoApp, and wrote the Windows Installer code.
I kind of know what I'm talking about. You're mixing up 2 things. Install location and Registry Location. Install location changed, the registry location didn't.

Thanks for reporting these information.

I know what's going on as we have a similar issues with other custom-built Firefox.

I appreciate the time you spent helping out. I will address the issues in an upcoming update.

**Comment by gaveitatry2** [comment]
The  next time that I install any version of VDH and CoApp, I'm going to manually delete all those registry keys. I don't know if it will make a difference, but I don't see why they are needed. A new install will just recreate them, and by deleting them beforehand I can remove some registry keys that are no longer needed, which will diminish some confusion and make things less cluttery.

**Comment by CodehunterWorks** [comment]
I dont think it is a registry problem because I'm on Linux Mint with Firefox and Tor Browser. VDH 8.1.4.0 cannot find CoApp. I think VDH 8.1.4.0 is generally broken with Firefox based Browsers.

**Comment by paulrouget** [comment]
> I dont think it is a registry problem because I'm on Linux Mint with Firefox and Tor Browser. VDH 8.1.4.0 cannot find CoApp. I think VDH 8.1.4.0 is generally broken with Firefox based Browsers.

There are some special configurations where we're having issues indeed. Linux Mint is one of them.

Here are my recommendations, and let me know if that helps or not:

Make sure to run the install command as non-root use:

```
 /opt/vdhcoapp/vdhcoapp install
```

Make sure you allow greenlight the coapp in flatpak:

```
sudo apt-get install -y flatpak
flatpak permission-set webextensions net.downloadhelper.coapp snap.firefox yes
```

Restart Firefox, and let me know if this helps.

**Comment by CodehunterWorks** [comment]
> ```
>  /opt/vdhcoapp/vdhcoapp install
> ...
> Restart Firefox, and let me know if this helps.

This command has fixed Firefox, bot not Tor Browser. As far as I can see in the script output, TB is not recognized, there is no word about TB. In my case, TB was not installed via dpkg or flatpak. It is simply a decompressed zip.

**Comment by gaveitatry2** [comment]
@paulrouget

I am not in a hurry to get this problem fixed, because I am using VDH 8.0.0.6 with CoApp v1 and it works.

I am also not in a hurry to try any new release or update because I don't want to encounter problems again, and I am too lazy to do any more testing for a while. I would rather wait until there is a solid, stable release where everyone says the problem has been fixed.

But I want to know if you will let me know once it is fixed by making a post in this Issue. I don't see a changelog that says "this problem has been fixed" and I don't want to assume something is fixed when it is not. So I am hoping you will post on this Issue once it has been fixed. Or if not, that's fine, but please let me know that you won't let me know so I won't be waiting forever. Thank you.

**Comment by paulrouget** [comment]
You'll get a github notification when this is fixed.

**Comment by gaveitatry2** [comment]
> You'll get a github notification when this is fixed.

Hi. It's been four months (Feb 7 but almost Jun 7 now). No hurry, but I was wondering if you still plan on figuring this out.

**Comment by gaveitatry2** [comment]
@paulrouget - I am going to try and make the switch from Mozilla Firefox to Google Chrome because Firefox feels outdated. I just realized that I had a VDH license for Firefox that I never really used. I meant to use it to convert files but I never needed to convert anything. Anyway, is there any way I can convert this license to Google Chrome so I won't have to buy it again?

**Comment by strykefreedom** [comment]
> [@paulrouget](https://github.com/paulrouget)
> 
> Yes, what you said is correct.
> 
> **THIS DOES NOT WORK:**
> 
> Mozilla Firefox (Portable) installed at: "G:\Portable Apps\PortableApps\FirefoxPortable"
> 
> Video DownloadHelper 8.1.4.0
> 
> CoApp 2.0.10 installed at (I tried both Windows 64-bit and 32-bit versions): "G:\Portable Apps\PortableApps\FirefoxPortable\DownloadHelper CoApp"
> 
> **THIS WORKS:**
> 
> Mozilla Firefox (Portable) installed at: "G:\Portable Apps\PortableApps\FirefoxPortable"
> 
> Video DownloadHelper 8.0.0.6
> 
> CoApp 1.6.3 installed at: "G:\Portable Apps\PortableApps\FirefoxPortable\net.downloadhelper.coapp"
> 
> The only difference between what works and does not work is the VDH and CoApp versions. The custom install location is the exact same in both scenarios except that the new CoApp folder is named differently than what it used to be named.
> 
> Also, no Antivirus blocking anything.
> 
> I am just going to use the older version until you figure things out.

HI gaveitatry2, I've tried to replicate your setup with the versions you've outlined in the working portable installation. I can't get the extension to detect the companion app on my usb or portable firefox directory. Were there any additional changes you made other than just installing into the directories?

**Comment by gaveitatry2** [comment]
I really have no clue. I am sorry. I have not used Mozilla Firefox (Portable) in almost a year. I switched to Google Chrome a long time ago. On a scale from 1 to 1 billion, 1 billion meaning perfect memory and 1 being no memory, my memory on how to make this work for Firefox is a 1. Your best bet is to ask paul for support because I am completely unable to help in any way. It's been too long.
